crml/transaction-payment/src/constants.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd. and Centrality Investments Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Transaction Payment Module
+//!
+//! Transaction Payment Customized Error Code Constants
+
+pub mod error_code {
+	// Core to asset exchange errors
+	pub const CORE_TO_ASSET_PRICE_NOT_ABOVE_ZERO: u8 = 190;
+	pub const CORE_TO_ASSET_PRICE_ABOVE_MAX_LIMIT: u8 = 191;
+	pub const INSUFFICIENT_BUYER_CORE_ASSET_BALANCE: u8 = 192;
+	// Asset to core exchange errors
+	pub const ASSET_TO_CORE_PRICE_NOT_ABOVE_ZERO: u8 = 193;
+	pub const ASSET_TO_CORE_PRICE_ABOVE_MAX_LIMIT: u8 = 194;
+	pub const INSUFFICIENT_BUYER_TRADE_ASSET_BALANCE: u8 = 195;
+	// Other errors
+	pub const ASSET_SALE_VALUE_NOT_ABOVE_ZERO: u8 = 196;
+	pub const SALE_VALUE_BELOW_REQUIRED_MINIMUM: u8 = 197;
+	pub const INSUFFICIENT_SELLER_CORE_ASSET_BALANCE: u8 = 198;
+	pub const BUY_AMOUNT_NOT_POSITIVE: u8 = 199;
+	pub const INSUFFICIENT_FEE_ASSET_BALANCE: u8 = 200;
+	pub const INVALID_ASSET_ID: u8 = 201;
+	pub const UNKNOWN_BUY_FEE_ASSET: u8 = 202;
+	pub const LIQUIDITY_TOO_LOW: u8 = 203;
+	pub const ASSET_CANNOT_SWAP_FOR_ITSELF: u8 = 204;
+	pub const INSUFFICIENT_CORE_ASSET_RESERVE: u8 = 205;
+
+	// Matches and converts crml-cennzx-spot module errors, such that
+	// they are propagated in crml-transaction-payment module
+	pub fn buy_fee_asset_error_msg_to_code(message: &'static str) -> u8 {
```

Comment:
If you had the enum (Referring to my previous comment), then you may just need to implement ToString or Display for your enum. Then you should always pass the enum entry between functions until you want to log or print it in which case you would convert them to strings. So I believe converting strings to enum entries (or constants) which is not a light function (in terms of performance) could be avoidable.

Reply:
This function converts string message in `DispatchError` into the appropriate `InvalidTransaction::Custom(u8)` error type